cdap-distributions/src/hdinsight/pageblob-configure.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+#
+# Copyright © 2016 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+#
+# Configure HDFS Pageblob support for CDAP HDInsight cluster
+#
+
+# Ambari constants
+AMBARICONFIGS_SH=/var/lib/ambari-server/resources/scripts/configs.sh
+AMBARIPORT=8080
+ACTIVEAMBARIHOST=headnodehost
+
+# Function definitions
+
+die() { echo "ERROR: ${*}"; exit 1; };
+
+checkHostNameAndSetClusterName() {
+    fullHostName=$(hostname -f)
+    echo "fullHostName=$fullHostName"
+    CLUSTERNAME=$(sed -n -e 's/.*\.\(.*\)-ssh.*/\1/p' <<< $fullHostName)
+    if [ -z "$CLUSTERNAME" ]; then
+        CLUSTERNAME=$(echo -e "import hdinsight_common.ClusterManifestParser as ClusterManifestParser\nprint ClusterManifestParser.parse_local_manifest().deployment.cluster_name" | python)
+        if [ $? -ne 0 ]; then
+            echo "[ERROR] Cannot determine cluster name. Exiting!"
+            exit 133
+        fi
+    fi
```

Comment:
So... I would probably replace this entire thing with two lines:
```
CLUSTERNAME=${CLUSTERNAME:-$(python -c 'import hdinsight_common.ClusterManifestParser as ClusterManifestParser; print ClusterManifestParser.parse_local_manifest().deployment.cluster_name')
[[ ${CLUSTERNAME} ]] || die "Cannot determine cluster name. Exiting!" 133
```
- Don't bother with checking if it's empty before assignment, the shell can do this internally
- Using echo to another command when not necessary (python errors go to stderr, not stdout)
- Use `die` function, since we have it
- Two less conditional checks